core/src/main/java/com/linecorp/armeria/common/stream/FilteredStreamMessage.java

```diff
@@ -116,53 +123,60 @@ public boolean isEmpty() {
     @Override
     public void subscribe(Subscriber<? super U> subscriber) {
         requireNonNull(subscriber, "subscriber");
-        delegate.subscribe(new FilteringSubscriber(subscriber, false));
+        delegate.subscribe(new FilteringSubscriber(subscriber, false),
+                           filteringSubscriptionOptions().toArray(EMPTY_OPTIONS));
     }
 
     @Override
-    public void subscribe(Subscriber<? super U> subscriber, boolean withPooledObjects) {
+    public void subscribe(Subscriber<? super U> subscriber, SubscriptionOption... options) {
         requireNonNull(subscriber, "subscriber");
-        delegate.subscribe(new FilteringSubscriber(subscriber, withPooledObjects), withPooledObjects);
-    }
+        requireNonNull(options, "options");
 
-    @Override
-    public void subscribe(Subscriber<? super U> subscriber, EventExecutor executor) {
-        requireNonNull(subscriber, "subscriber");
-        requireNonNull(executor, "executor");
-        delegate.subscribe(new FilteringSubscriber(subscriber, false), executor);
+        delegate.subscribe(new FilteringSubscriber(subscriber, containsWithPooledObjects(options)),
+                           filteringSubscriptionOptions(options).toArray(EMPTY_OPTIONS));
     }
 
     @Override
     public void subscribe(Subscriber<? super U> subscriber, EventExecutor executor,
-                          boolean withPooledObjects) {
+                          SubscriptionOption... options) {
         requireNonNull(subscriber, "subscriber");
         requireNonNull(executor, "executor");
-        delegate.subscribe(new FilteringSubscriber(subscriber, withPooledObjects), executor, withPooledObjects);
+        requireNonNull(options, "options");
+        delegate.subscribe(new FilteringSubscriber(subscriber, containsWithPooledObjects(options)),
+                           executor, filteringSubscriptionOptions(options).toArray(EMPTY_OPTIONS));
     }
 
-    @Override
-    public CompletableFuture<List<U>> drainAll() {
-        return drainAll(false);
+    private ArrayList<SubscriptionOption> filteringSubscriptionOptions(SubscriptionOption... options) {
```

Comment:
Could just return `SubscriptionOption[]`? All callers convert it to an array.

Reply:
I think normally we'd add an `Iterable` override to `subscribe` instead of having callers always pass in an array. Would that make more sense? Since we can use an `EnumSet`, delegating things to an `Iterable` should be quite cheap too.

Reply:
I didn't add those variants to reduce the number of public APIs. But I can add them if we need.

Reply:
Yeah it could be overkill - Java APIs like `Files` usually don't have an iterator. It mainly came to mind since things like optimizing `containsX` below becomes much easier if using `EnumSet`. If it's possible to have it only in the internal API that would work too, but no worries if it seems too complicated.

Reply:
It's gonna be just a two-element array, so I guess delegating things to an `Iterable` is probably an overkill?

Reply:
Changed to return `SubscriptionOption[]`, thanks!

> `containsX` below becomes much easier if using `EnumSet`

Yeah, that made me almost to add the `Iterable` version. :-)

> If it's possible to have it only in the internal API that would work too, but no worries if it seems too complicated.

Yeah, it seems complicated as you said. :-) I will keep that in my mind, and add it when it's necessary.